### PR TITLE
feat(details): updated styling

### LINKED ISSUE #2036: ebay-details: update visual style
## Purpose
We're updating the ebay-details component (aka _Expansion_) to increase the title element's weight to bold and updating the text and icon color to `foreground.primary`. This increases it's contrast with surrounding content and aligns closer to Evo.

## References
[Project brief →](https://www.figma.com/file/48oQSeB23kDibHz9WB1coh/Expansion?node-id=1-58&t=2BzgPEqriX4cgzW5-4)

[Figma reference →](https://www.figma.com/file/48oQSeB23kDibHz9WB1coh/Expansion?node-id=531-64263&t=2BzgPEqriX4cgzW5-11)

[Skin docs →](https://opensource.ebay.com/skin/#details)

## Before/After
image

image

## Specs
image

### PR DESCRIPTION
**Pull request by agliga**
Fixes #2036

- [X] This PR contains CSS changes
- [ ] This PR does not contain CSS changes

## Description
* Updated details visual style
* Fixed icon layout (for old icon it should still align)

## Notes
* Design wants width to be fixed to item. For now, I left it as is and have the width span the whole content of the button. This is in order to prevent possible layout shift if we have things on the left/right of it
This is what it looks like if we remove width 100% (current not in this change)
Screenshot 2024-07-29 at 11 01 45 AM
With width 100%
Screenshot 2024-07-29 at 10 45 03 AM

## Screenshots
Old icon (12px)
Screenshot 2024-07-29 at 10 56 11 AM
New Icon (16px)
Screenshot 2024-07-29 at 10 56 56 AM

## Checklist
- [X] I verify the build is in a non-broken state
- [X] I verify all changes are within scope of the linked issue

- [X] I regenerated all CSS files under dist folder
- [X] I tested the UI in all supported browsers
- [ ] I did a visual regression check of the components impacted by doing a Percy build and approved the build
- [X] I tested the UI in dark mode and RTL mode

- [X] I added/updated/removed Storybook coverage as appropriate

### CODE REVIEW

**Comment by agliga** [comment]
Removed with:100. 
Only kept it for centered version (its needed otherwise it wont center it)
Also I have changed the inline-block to be inline flex. This properly centers the chevron and the text.

**Comment by agliga** [comment]
> Looks great!
> 
> Maybe there was some discussion to ditch the styling for states, but the issue currently has those styles with no comment about not using them. Shouldn't the component have those implemented?
> 
> image

I don't see these in the figma link which is why I didnt implement them.

File: src/sass/details/details.scss [hunk 1]
```diff
@@ -45,7 +54,7 @@ span.details__icon {
 
 /* progressive enhancement - override hidden SVG */
 span.details__icon[hidden] {
-    display: inline-block;
+    display: inline-flex;
```
[the remark this answers is not in the document]
Comment (ianmcburnie) [reply src/sass/details/details.scss:57]: Ignore that, I see this is on the icon. My bad.

**Comment by ianmcburnie** [comment]
Given that we removing the 100% width, does this centered variant make sense anymore? 

Screenshot 2024-07-30 at 10 15 21 AM

**Comment by agliga** [comment]
> Given that we are removing the 100% width on the button/summary, does this centered variant make sense anymore?
> 
> I would say that this modifier is no longer just centering the button text anymore, it is now centering and making the button fluid: https://github.com/eBay/skin/pull/2403/files#diff-df3ce4e1c4141c74932c4e28e5175bb81d45973c23a752a8ec55e5d7bd11d540R43
> 
> Screenshot 2024-07-30 at 10 15 21 AM

Its not! I think design removed it. We can remove it from docs, but I kept it in for now until we get clarification. 
However, changing this would be breaking.

**Review by ArtBlue — APPROVED** [review]
LGTM!

**Comment by ianmcburnie** [comment]
> Its not! I think design removed it. We can remove it from docs, but I kept it in for now until we get clarification.
However, changing this would be breaking.

Ok. Let's remove from docs and added a deprecated comment in the code.

**Pull request merged**